Automatic state transition

MirkoTV opened the pull request:
Automatic meeting state transition 

#77

MirkoTV committed:
EF-77: automatic_state_transition

- Using changeMeetingStateEvent in general.

MirkoTV committed:
EF-77: automatic_state_transition

- Updating meeting dates to support automatic transition.

MirkoTV committed:
EF-77: automatic_state_transition

- Adding quartz dependency
- Adding a job to make automatic transition of states when needed

MirkoTV committed:
EF-77: automatic_state_transition

- Fixing conflicts from merging from master

MirkoTV committed:
Merge branch 'master' into automatic_state_transition

Conflicts:
	pom.xml
	src/main/java/org/jala/efeeder/foodmeeting/FoodMeetingManager.java
	src/main/webapp/assets/js/voting.js

MirkoTV committed:
EF-77: automatic_state_transition

- Fix minor bugs

MirkoTV committed:
EF-77: automatic_state_transition

- Fix error in javascript meeting room name

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/ChangeFoodMeetingStatusCommand.java:
```diff
@@ -35,18 +38,19 @@ public Out execute(In parameters) throws Exception {
 		);
-		MessageContext messageContext = MessageContext.newBuilder()
-				.setRoom(receivedMessage.getRoom())
+		
```
remove tabs

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/ChangeFoodMeetingStatusCommand.java:
```diff
@@ -35,18 +38,19 @@ public Out execute(In parameters) throws Exception {
-				.setEvents(events)
-				.build();
+				.setEvents(events);
+		
```
remove tabs

MirkoTV replied:
I think this one is necessary to separate code blocks that are related to improve readability.

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/EditFoodMeetingCommand.java:
```diff
@@ -24,7 +24,7 @@
 public class EditFoodMeetingCommand implements CommandUnit {
 
-	private static final String UPDATE_FOOD_MEETING_SQL = "UPDATE food_meeting SET name= ?, image_link= ?, status=?, event_date=?, "
+	private static final String UPDATE_FOOD_MEETING_SQL = "UPDATE food_meeting SET name= ?, image_link= ?, event_date=?, "
```
move to use a manager

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java:
```diff
@@ -80,4 +80,22 @@ public String getTime() {
 		DateTimeFormatter fmt = DateTimeFormat.forPattern("HH:mm");
 		return fmt.print(eventDate.getTime());
 	}
+	
```
remove tabs

MirkoTV replied:
I think this one is necessary to separate the methods

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java:
```diff
@@ -80,4 +80,22 @@ public String getTime() {
+		if(time.compareTo(this.eventDate) >= 0) {
+			state = FoodMeetingStatus.Finish;
+		}
+		
```
remove tabs

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/FoodMeeting.java:
```diff
@@ -80,4 +80,22 @@ public String getTime() {
+	
+	public FoodMeetingStatus getStatusByTime(Timestamp time) {
+		FoodMeetingStatus state = FoodMeetingStatus.Voting;
+		if(time.compareTo(this.votingDate) >= 0) {
```
change the if conditions to this.votingDate.compareTo(time), because you know that this.votingDate is not null, but you don't know if the time can be null

rayaviri commented on src/main/java/org/jala/efeeder/foodmeeting/FoodMeetingManager.java:
```diff
@@ -27,22 +33,36 @@ public FoodMeeting getFoodMeetingById(int id) throws SQLException {
 		preparedStatement.setInt(3, idUser);
 		preparedStatement.executeUpdate();
 	}
+	
```
remove tabs

MirkoTV replied:
I think this one is necessary to separate the methods

rayaviri commented on src/main/java/org/jala/efeeder/scheduledJobs/ChangeMeetingStateJob.java:
```diff
@@ -0,0 +1,102 @@
+ * @author Mirko Terrazas
+ */
+public class ChangeMeetingStateJob implements org.quartz.Job {
+	
```
review the all tabs  and delete them,

rayaviri commented on src/main/java/org/jala/efeeder/servlets/CommandEndpoint.java:
```diff
@@ -98,4 +98,16 @@ public void onMessage(String message, Session session) {
+	
+	public static boolean sendMessage(MessageContext message) {
+		boolean isRoomEmpty = true;
+		
```
remove tabs

rayaviri commented on src/main/java/org/jala/efeeder/servlets/CommandEndpoint.java:
```diff
@@ -98,4 +98,16 @@ public void onMessage(String message, Session session) {
+		if(!isRoomEmpty) {
+			foodMeetingRoom.sendMessage(message);
+		}
+		
```
remove tabs

rayaviri commented on src/main/java/org/jala/efeeder/servlets/StartUpServlet.java:
```diff
@@ -0,0 +1,59 @@
+public class StartUpServlet extends HttpServlet {
+	
+	private Scheduler scheduler = null;
+	
```
remove the tabs

rayaviri commented on src/main/java/org/jala/efeeder/servlets/StartUpServlet.java:
```diff
@@ -0,0 +1,59 @@
+
+			scheduler.scheduleJob(changeMeetingStateJob, changeMeetingStateTrigger);
+		} catch (SchedulerException ex) {
+			Logger.getLogger(StartUpServlet.class.getName()).log(Level.SEVERE, "Error when trying to start quartz jobs scheduler.", ex);
```
use the log4j

rayaviri commented on src/main/java/org/jala/efeeder/servlets/StartUpServlet.java:
```diff
@@ -0,0 +1,59 @@
+		try {
+			scheduler.shutdown();
+		} catch (SchedulerException ex) {
+			Logger.getLogger(StartUpServlet.class.getName()).log(Level.SEVERE, "Error trying to shutdown quartz jobs scheduler", ex);
```
use the log4j

rayaviri commented on src/main/java/org/jala/efeeder/scheduledJobs/ChangeMeetingStateJob.java:
```diff
@@ -0,0 +1,102 @@
+				}
+			});			
+		} catch (SQLException ex) {
+			Logger.getLogger(ChangeMeetingStateJob.class.getName()).log(Level.SEVERE, null, ex);
```
use log4j

rayaviri commented on src/main/java/org/jala/efeeder/suggestion/SetWinnerPlaceCommand.java:
```diff
@@ -54,12 +56,20 @@ public Out execute(In parameters) throws Exception {
+						.setNewStatus(FoodMeetingStatus.Order.name())
+						.build())
+					.build());
+			
```
remove tabs

MirkoTV replied:
I Think this space helps to separate the events creation from the message creation, and helps for readability

rayaviri commented on src/main/java/org/jala/efeeder/util/DateFormatter.java:
```diff
@@ -14,4 +15,8 @@
 	public static String format(Timestamp date) {
 		return SIMPLE_FORMAT.format(date);
 	}
+	
```
remove tabs

MirkoTV replied:
I think this one is necessary to separate the methods

rayaviri commented on src/main/webapp/assets/js/communication-service.js:
```diff
@@ -277,7 +265,11 @@
       }
     };
-
+  
```
remove white space

MirkoTV replied:
I think this on is necessary to separate the methods

rayaviri commented on src/main/webapp/assets/js/foodMeetings.js:
```diff
@@ -1,6 +1,6 @@
 $(function() {
 
-	var createMeetingRoomId = "createMeetingRoomId";
+	var homeRoomId = "homeRoom";
```
I think that you can move the all logic into other method?

MirkoTV committed:
EF-77: automatic_state_transition

- Fix code review observations
- Fix error when meeting is pretty close y meeting slider not rendering correctly

rayaviri reviewed (commented):
coments

MirkoTV committed:
EF-77: automatic_state_transition

- Moving update meeting to the foodMeetingManager
- Applying log4j
- Refactor foodMeetings.js

Other:

- Fixing typo in log4j properties file name

MirkoTV commented on src/main/java/org/jala/efeeder/foodmeeting/ChangeFoodMeetingStatusCommand.java:
```diff
@@ -35,18 +38,19 @@ public Out execute(In parameters) throws Exception {
-				.setEvents(events)
-				.build();
+				.setEvents(events);
+		
```
I think this one is necessary to separate block of codes that are related to improve readability.

MirkoTV committed:
EF-77: automatic_state_transition

- Removed empty lines

MirkoTV committed:
EF-77: automatic_state_transition

- Removed extra tabs

rayaviri reviewed (approved):
look fine

MirkoTV committed:
EF-77: automatic_state_transition

- Fixed error when refactoring in times comparision
- Moving logger config to INFO mode
